pallets/nfts/src/features/approvals.rs

```diff
@@ -165,10 +179,197 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 
 		Self::deposit_event(Event::AllApprovalsCancelled {
 			collection,
-			item,
+			item: Some(item),
 			owner: details.owner,
 		});
 
 		Ok(())
 	}
+
+	/// Approves the transfer of items in the collection that owned by the origin to a delegate.
+	///
+	/// This function is used to approve the transfer of items in the `collection` that owned by the
+	/// `origin` to a `delegate`. The `delegate` is the account that will be allowed to take control
+	/// of items in the collection. Optionally, a `deadline` can be specified to set a time limit
+	/// for the approval. The `deadline` is expressed in block numbers and is added to the current
+	/// block number to determine the absolute deadline for the approval. After approving the
+	/// transfer, the function emits the `TransferApproved` event.
+	///
+	/// This function reserves the required deposit from the `origin` account. If an approval
+	/// already exists, the new amount is added to such existing approval.
```

Comment:
This is not clear to me, "already exists" seems to me that we are approving the same delegate again.

Reply:
Reference from this: https://github.com/paritytech/polkadot-sdk/blob/445c1c80d438d5b937c347b04e4daa66ad25d878/substrate/frame/assets/src/functions.rs#L895

If the approve already exists, we ask for more deposits from the user. User can override an existing approval, for example, changing the deadline of the approval.

Reply:
Can we improve this sentence then please